File: module4/_01_overview_spring_mvc/exercise/dictionary/src/main/java/service/DictionaryService.java
package service;

import java.util.HashMap;
import java.util.Map;

public class DictionaryService {
    static Map<String, String> dictionary = new HashMap<>();

    static {
        dictionary.put("hello", "xin chao");
        dictionary.put("beautiful", "xinh dep");
        dictionary.put("world", "the gioi");
        dictionary.put("red", "màu đỏ");
    }

    public String search(String val) {
        String result= dictionary.get(val);
        return result;
    }
}

File: module5/_07_service_router/exercise/e1/dictionary/src/app/service/dictionary.service.ts
import { Injectable } from '@angular/core';
import {Dictionary} from "../dictionary";

@Injectable({
  providedIn: 'root'
})
export class DictionaryService {
  words : Dictionary[] = [
    {
      word: "hello",
      mean: "xin chao"
    },
    {
      word: "world",
      mean: "the gioi"
    },
    {
      word: "computer",
      mean: "may tinh"
    }
  ]


  constructor() { }

  findAll(){
    return this.words;
  }

  find(word: string) : any{
    return this.words.find(e => {
      return  e.word == word;
    })
  }
}
